LBH.AdultSocialCare.Api/V1/Gateways/Common/Concrete/CarePackageHistoryGateway.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LBH.AdultSocialCare.Api.V1.Domain.Common;
+using LBH.AdultSocialCare.Api.V1.Gateways.Common.Interfaces;
+using LBH.AdultSocialCare.Api.V1.Infrastructure;
+using Microsoft.EntityFrameworkCore;
+
+namespace LBH.AdultSocialCare.Api.V1.Gateways.Common.Concrete
+{
+    public class CarePackageHistoryGateway : ICarePackageHistoryGateway
+    {
+        private readonly DatabaseContext _dbContext;
+
+        public CarePackageHistoryGateway(DatabaseContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+        public async Task<IEnumerable<CarePackageHistoryDomain>> ListAsync(Guid carePackageId)
+        {
+            var carePackageHistory = await _dbContext.CarePackageHistories
+                .Where(cph => cph.CarePackageId.Equals(carePackageId))
+                .Select(cph => new CarePackageHistoryDomain
```

Comment:
In future implementation you can decorate your entity with 

- [GenerateMapping(typeof(CarePackageHistoryDomain))] 
- [GenerateListMapping(typeof(CarePackageHistoryDomain))] 
 
and mapping methods will be generated automatically for you, just build project